From 700d2121b1dc826cca412e03fb6326daa531d507 Mon Sep 17 00:00:00 2001
From: nxa22940 <xiaodong.zhang@nxp.com>
Date: Fri, 21 Apr 2023 14:11:15 +0800
Subject: [PATCH] CMakeLists.txt: remove fmt version

Signed-off-by: nxa22940 <xiaodong.zhang@nxp.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 44e62b1..4229194 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ project(maliput LANGUAGES C CXX VERSION 3.0.0)
 message(STATUS "\n\n====== Finding 3rd Party Packages ======\n")
 
 find_package(ament_cmake REQUIRED)
-find_package(fmt 6.1.2 EXACT REQUIRED)
+find_package(fmt EXACT REQUIRED)
 find_package(gflags REQUIRED)
 find_package(yaml-cpp REQUIRED)
 
-- 
2.35.1

