From 91258b35d80ae66ef41504e9eaf47e0a36581a32 Mon Sep 17 00:00:00 2001
From: nxa22940 <xiaodong.zhang@nxp.com>
Date: Thu, 17 Aug 2023 10:27:51 +0800
Subject: [PATCH] Check input parameter of getCost when get cost from costmap

Signed-off-by: nxa22940 <xiaodong.zhang@nxp.com>
---
 src/costmap_2d.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/costmap_2d.cpp b/src/costmap_2d.cpp
index 3d50327b1..566fc1a40 100644
--- a/src/costmap_2d.cpp
+++ b/src/costmap_2d.cpp
@@ -36,6 +36,7 @@
  *         David V. Lu!!
  *********************************************************************/
 #include <costmap_2d/costmap_2d.h>
+#include <costmap_2d/cost_values.h>
 #include <cstdio>
 
 using namespace std;
@@ -191,12 +192,16 @@ unsigned char* Costmap2D::getCharMap() const
 
 unsigned char Costmap2D::getCost(unsigned int mx, unsigned int my) const
 {
-  return costmap_[getIndex(mx, my)];
+  if (mx <= size_x_ && my <= size_y_)
+    return costmap_[getIndex(mx, my)];
+  else
+    return NO_INFORMATION;
 }
 
 void Costmap2D::setCost(unsigned int mx, unsigned int my, unsigned char cost)
 {
-  costmap_[getIndex(mx, my)] = cost;
+  if (mx <= size_x_ && my <= size_y_)
+    costmap_[getIndex(mx, my)] = cost;
 }
 
 void Costmap2D::mapToWorld(unsigned int mx, unsigned int my, double& wx, double& wy) const
-- 
2.35.1

